Patterns/Chinese/Chinese-NumbersWithUnit.yaml

```diff
@@ -756,4 +756,6 @@ TemperatureAmbiguousValues: !list
   entries:
     - 度
     - k
+HalfUnitRegex: !simpleRegex
+  def: 半
```

Comment:
Shouldn't this regex check if the match is at the end of the string or similar?

Reply:
Due to the different regex results in Java, we need to match the "半" pattern from the source string rather than the pre-recognizer phrases. And the index check makes sure the "半" immediately follows a phrase which composed of a number and a unit.